models/repo_indexer.go

```diff
@@ -232,6 +232,19 @@ func addDelete(filename string, repo *Repository, batch rupture.FlushingBatch) e
 }
 
 func isIndexable(entry *git.TreeEntry) bool {
+	if setting.Indexer.FilePatterns != nil {
+		var found bool
+		name := strings.ToLower(entry.Name())
+		for _, g := range setting.Indexer.FilePatterns {
+			if g.Match(name) {
+				found = true
+				break
+			}
+		}
+		if found != setting.Indexer.IncludePatterns {
+			return false
+		}
+	}
 	return entry.IsRegular() || entry.IsExecutable()
```

Comment:
this check should probably go before file pattern check. Also `if setting.Indexer.FilePatterns != nil {` should be reversed for cleaner code, something like:

```go
func isIndexable(entry *git.TreeEntry) bool {
	if !entry.IsRegular() && !entry.IsExecutable() {
		return false
	}
	if setting.Indexer.FilePatterns == nil {
		return true
	}
	// Check pattern match
	var found bool
	...
```

Reply:
OK, I'll turn that the other way around. About the new vendor, ¿should I simply "make vendor" and add anything that's modified in a new commit?

Reply:
Done!